src/InsertionsClient/Api/Props/PropsVariableDeducer.cs

```diff
@@ -0,0 +1,175 @@
+            {
+                return await GetPackageFileList(defaultConfigUpdater, packageUpdate);
+            });
+
+            ActionBlock<string> filenameMatchBlock = new ActionBlock<string>((filename) =>
+            {
+                foreach (PropsFileVariableReference varReference in MatchFileNames(swrFiles, filename))
+                {
+                    deducedVariables.Add(varReference);
+                }
+            });
+
+            nugetDownloadBlock.LinkTo(filenameMatchBlock, new DataflowLinkOptions() { PropagateCompletion = true });
+
+            foreach (PackageUpdateResult package in updatedPackages)
+            {
+                nugetDownloadBlock.Post(package);
+            }
+
+            nugetDownloadBlock.Complete();
+            filenameMatchBlock.Completion.Wait();
+
+            return deducedVariables.ToList();
+        }
+
+        private async Task<IEnumerable<string>> GetPackageFileList(DefaultConfigUpdater defaultConfigUpdater, PackageUpdateResult packageUpdate)
+        {
+            string? link = defaultConfigUpdater.GetPackageLink(packageUpdate.PackageId);
+
+            if (link == null)
+            {
+                // We don't know where nuget package should be extracted to.
+                return Enumerable.Empty<string>();
+            }
+
+            SourceCacheContext cache = new SourceCacheContext();
+            CancellationTokenSource cancellationToken = new CancellationTokenSource();
+            NuGetVersion packageVersion = new NuGetVersion(packageUpdate.NewVersion);
+
+            // Stream that will receive the package bytes
+            using MemoryStream packageStream = new MemoryStream();
+
+            try
+            {
+                SourceRepository? rep = Repository.Factory.GetCoreV3(_feed, FeedType.HttpV3);
+                if (_credentials != null)
+                {
+                    rep.PackageSource.Credentials = _credentials;
+                }
+
+                FindPackageByIdResource? resource = await rep.GetResourceAsync<FindPackageByIdResource>();
+
+                Trace.WriteLine($"Downloading package {packageUpdate.PackageId}-{packageUpdate.NewVersion} into memory.");
+
+                bool downloadResult = await resource.CopyNupkgToStreamAsync(
+                    packageUpdate.PackageId,
+                    packageVersion,
+                    packageStream,
+                    cache,
+                    NullLogger.Instance,
+                    cancellationToken.Token);
+
+                if (!downloadResult)
+                {
+                    Trace.WriteLine($"There is an issue downloading nuget package: {packageUpdate.PackageId}");
+                    return Enumerable.Empty<string>();
+                }
+
+                Trace.WriteLine($"Downloading complete for package {packageUpdate.PackageId}.");
+
+                using PackageArchiveReader packageReader = new PackageArchiveReader(packageStream);
```

Comment:
do you think it would improve maintainability if this method were modulized a bit?

Reply:
Separated file name extraction into a separate method.